cmake/Modules/FindTPLONEDPL.cmake

```
@@ -0,0 +1,4 @@
+FIND_PACKAGE(oneDPL)
+IF (oneDPL_FOUND)
+  KOKKOS_CREATE_IMPORTED_TPL(ONEDPL INTERFACE LINK_LIBRARIES oneDPL)
+ENDIF()
```

When you use an `oneAPI` installation, it doesn't provide a find module but still includes the headers and libraries.
Also, we should require finding the package here if necessary. Thus, I would do something like
```suggestion
INCLUDE(CheckIncludeFileCXX)
CHECK_INCLUDE_FILE_CXX(oneapi/dpl/execution KOKKOS_COMPILER_HAS_ONEDPL_EXECUTION_HEADER)
CHECK_INCLUDE_FILE_CXX(oneapi/dpl/algorithm KOKKOS_COMPILER_HAS_ONEDPL_ALGORITHM_HEADER)

IF (KOKKOS_COMPILER_HAS_ONEDPL_EXECUTION_HEADER AND KOKKOS_COMPILER_HAS_ONEDPL_ALGORITHM_HEADER)
  KOKKOS_CREATE_IMPORTED_TPL(ONEDPL INTERFACE)
ELSE()
  FIND_PACKAGE(oneDPL REQUIRED)
  KOKKOS_CREATE_IMPORTED_TPL(ONEDPL INTERFACE LINK_LIBRARIES oneDPL)
ENDIF()
```
instead.

---

Done. Thanks for that!